### DISCUSSION TITLE
Add default content checklist items to all custom post types

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/publishpress-checklists#450] [Feature Requests]
> I concur that this one is very dependent on the particular website PP is
> being used on. And I understand that by default, most of the checklist
> items might not be relevant to completely unknown CPT the user might have.
> But for us, most of the post types are all valid types of pretty much the
> same kind of content with different visualizations. Therefore I wanted to
> have the default Content Checklist items for all the custom post types.
> 
> The attached patch does that in a pretty uncool (read unsafe) manner, but it does the job for
> me. If you were to implement it better, I think this belongs either as a UI
> option in the admin panel, or *better yet *as a filter that collects the
> CPT the user wants the default checklist items to be accessible for. 
> 
> *Note: this patch is for the* *Content Checklist *addon, not for PP itseld.

Thanks to @jorr for the report and patch

[0003-Joro-PublishPress-Content-Checklist-add-default-chec.patch.zip](https://github.com/publishpress/PublishPress/files/2841797/0003-Joro-PublishPress-Content-Checklist-add-default-chec.patch.zip)

**Comment by jorr** [comment]
@stevejburge @andergmartins should I rework this as a PR?

**Comment by andergmartins** [comment]
@jorr please, add a pull request 👍 
Tks for the help.

**Comment by jorr** [comment]
@andergmartins on a second thought, I think a filter that allows users to add CPTs for each content checklist item might be a better idea, or do you think that's too much?

**Comment by andergmartins** [comment]
Hi @jorr we have a filter available for that: publishpress_checklists_post_types.
Our plugin uses it to populate the list of post types using the following criteria:

- post: always
- pages: always
- all post types queried using _builtin = false and public = true.

But you should be able to inject your own post types using that filter. Just make sure to use a priority higher than 10.
